translator/util/ec2util/ec2util.go

```diff
@@ -37,39 +40,69 @@ func initEC2UtilSingleton() (newInstance *ec2Util) {
 		return
 	}
 
-	ses, e := session.NewSession()
-	if e != nil {
-		log.Println("E! [EC2] getting new session info: ", e)
+	// Need to account for the scenario where a user running the CloudWatch agent on-premises,
+	// and doesn't require connectivity with the EC2 instance metadata service, while still
+	// gracefully waiting for network access on EC2 instances.
+	networkUp := false
+	for retry := 0; !networkUp && retry < allowedRetries; retry++ {
+		ifs, err := net.Interfaces()
+
+		if err != nil {
+			log.Println("E! [EC2] An error occurred while fetching network interfaces: ", err)
+		}
+
+		for _, in := range ifs {
+			if (in.Flags&net.FlagUp) != 0 && (in.Flags&net.FlagLoopback) == 0 {
```

Comment:
I was going to say I don't like the style of "no space around the ampersand (&)", but that is probably just what gofmt does so I can't complain.

Reply:
I didn't like it either, but this is what Goland was doing by default. I can update it. Not sure if `gofmt` would revert it, so I'll check that too I guess

Reply:
`go fmt ./translator/util/ec2util/` removed the spaces that I reintroduced, so I guess we're stuck unless we want to override that specific behavior. Not sure if it's worth doing

Reply:
If we do a lot of setting and checking of bits we could add helper functions like:
https://yourbasic.org/golang/bitmask-flag-set-clear/
Not suggesting it needs to be done now, just keep it in mind.